### PR TITLE
Add hannah65 PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description
Add hannah65 PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/hannah65/hannah65.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+#pragma once
+
+#include "quantum.h"
+#ifdef KEYBOARD_mechlovin_hannah65_rev1
+    #include "rev1.h"
+#elif KEYBOARD_mechlovin_hannah65_mechlovin9
+    #include "mechlovin9.h"
+#endif
```
Comment (drashna) [inline keyboards/mechlovin/hannah65/hannah65.h:24]:
```suggestion
#if defined(KEYBOARD_mechlovin_hannah65_rev1)
#    include "rev1.h"
#elif defined(KEYBOARD_mechlovin_hannah65_mechlovin9)
#    include "mechlovin9.h"
#endif
```

File: keyboards/mechlovin/hannah65/rev1/info.json [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                {"label":"K42 (B4,B7)", "x":2.5, "y":4, "w":1.25},
+                {"label":"K46 (B4,C3)", "x":3.75, "y":4, "w":6.25},
+                {"label":"K4A (B4,C7)", "x":10, "y":4, "w":1.25},
+                {"label":"K4B (B4,D0)", "x":11.25, "y":4, "w":1.25},
+                {"label":"K4C (B4,D1)", "x":13, "y":4},
+                {"label":"K4D (B4,D2)", "x":14, "y":4},
+                {"label":"K4E (B4,D3)", "x":15, "y":4}
+            ]
+        },
+    }
```
Comment (drashna) [inline keyboards/mechlovin/hannah65/rev1/info.json:224]: I think this line might be extra (PR Lint has a fit on/near this line

File: keyboards/mechlovin/hannah65/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/mechlovin/hannah65/rules.mk:21]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/hannah65/mechlovin9/readme.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# mechlovin9
+
+![mechlovin9](https://i.imgur.com/T5Lm6OY.png)
+
+A 65% PCB, compatible with Kyuu.
+
+
```
Comment (fauxpark) [inline keyboards/mechlovin/hannah65/mechlovin9/readme.md:7]:
```suggestion
```

File: keyboards/mechlovin/hannah65/readme.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# hannah65
+
+![hannah65](https://i.imgur.com/gallery/HdguPHL)
+
+A 65% PCB, compatible with E6.5, Kyuu,...
+
+
```
Comment (fauxpark) [inline keyboards/mechlovin/hannah65/readme.md:7]:
```suggestion
```

File: keyboards/mechlovin/hannah65/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/mechlovin/hannah65/rules.mk:19]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/mechlovin/hannah65/rev1/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  KC_HOME,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                    KC_PGDN,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT,          KC_UP,   KC_END,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
```
Comment (fauxpark) [inline keyboards/mechlovin/hannah65/rev1/keymaps/via/keymap.c:26]: You need to add three blank layers here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/hannah65/rev1/rev1.h [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    K40, K41, K42,               K46,                K4A, K4B,      K4C, K4D, K4E    \
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,   K1E   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,   K2E   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   K3E   }, \
+    { K40,   K41,   K42,   KC_NO, KC_NO, KC_NO, K46,   KC_NO, KC_NO, KC_NO, K4A,   K4B,   K4C,   K4D,   K4E   }, \
+}
+
+#define LAYOUT_ansi( \
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/rev1/rev1.h:35]:
```suggestion
#define LAYOUT_65_ansi_blocker( \
```

File: keyboards/mechlovin/hannah65/rev1/rev1.h [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    K40,  K41,  K42,              K46,                K4A,  K4B,     K4C, K4D, K4E    \
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   KC_NO, K1E   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,   K2E   }, \
+    { K30,   KC_NO, K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   K3E   }, \
+    { K40,   K41,   K42,   KC_NO, KC_NO, KC_NO, K46,   KC_NO, KC_NO, KC_NO, K4A,   K4B,   K4C,   K4D,   K4E   }, \
+}
+
+#define LAYOUT_iso( \
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/rev1/rev1.h:49]:
```suggestion
#define LAYOUT_65_iso_blocker( \
```

File: keyboards/mechlovin/hannah65/rev1/info.json [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+                {"label":"K42 (B4,B7)", "x":2.5, "y":4, "w":1.25},
+                {"label":"K46 (B4,C3)", "x":3.75, "y":4, "w":6.25},
+                {"label":"K4A (B4,C7)", "x":10, "y":4, "w":1.25},
+                {"label":"K4B (B4,D0)", "x":11.25, "y":4, "w":1.25},
+                {"label":"K4C (B4,D1)", "x":13, "y":4},
+                {"label":"K4D (B4,D2)", "x":14, "y":4},
+                {"label":"K4E (B4,D3)", "x":15, "y":4}
+            ]
+        },
+        "LAYOUT_ansi": {
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/rev1/info.json:81]:
```suggestion
        "LAYOUT_65_ansi_blocker": {
```

File: keyboards/mechlovin/hannah65/rev1/info.json [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+                {"label":"K42 (B4,B7)", "x":2.5, "y":4, "w":1.25},
+                {"label":"K46 (B4,C3)", "x":3.75, "y":4, "w":6.25},
+                {"label":"K4A (B4,C7)", "x":10, "y":4, "w":1.25},
+                {"label":"K4B (B4,D0)", "x":11.25, "y":4, "w":1.25},
+                {"label":"K4C (B4,D1)", "x":13, "y":4},
+                {"label":"K4D (B4,D2)", "x":14, "y":4},
+                {"label":"K4E (B4,D3)", "x":15, "y":4}
+            ]
+        },
+        "LAYOUT_iso": {
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/rev1/info.json:152]:
```suggestion
        "LAYOUT_65_iso_blocker": {
```

File: keyboards/mechlovin/hannah65/rev1/info.json [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+                {"label":"K42 (B4,B7)", "x":2.5, "y":4, "w":1.25},
+                {"label":"K46 (B4,C3)", "x":3.75, "y":4, "w":6.25},
+                {"label":"K4A (B4,C7)", "x":10, "y":4, "w":1.25},
+                {"label":"K4B (B4,D0)", "x":11.25, "y":4, "w":1.25},
+                {"label":"K4C (B4,D1)", "x":13, "y":4},
+                {"label":"K4D (B4,D2)", "x":14, "y":4},
+                {"label":"K4E (B4,D3)", "x":15, "y":4}
+            ]
+        },
+    }
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/rev1/info.json:224]:
```suggestion
        }
    }
}
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10284/files).

**Comment by noroadsleft** [comment]
[One review comment outstanding.](https://github.com/qmk/qmk_firmware/pull/10284#discussion_r493902167)

File: keyboards/mechlovin/hannah65/mechlovin9/info.json [hunk 13]
```diff
@@ -0,0 +1,80 @@
+{
+    "keyboard_name": "mechlovin9",
+    "url": "",
+    "maintainer": "Team Mechlovin",
+    "width": 16,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
+            "layout_all": [
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/mechlovin9/info.json:9]:
```suggestion
        "LAYOUT_all": {
            "layout": [
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by mechlovin** [comment]
> GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10284/files).

Its new for me. Thank you

File: keyboards/mechlovin/hannah65/mechlovin9/info.json [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+                {"label":"K25 (B2,C2)", "x":5.75, "y":2},
+                {"label":"K26 (B2,C3)", "x":6.75, "y":2},
+                {"label":"K27 (B2,C4)", "x":7.75, "y":2},
+                {"label":"K28 (B2,C5)", "x":8.75, "y":2},
+                {"label":"K29 (B2,C6)", "x":9.75, "y":2},
+                {"label":"K2A (B2,C7)", "x":10.75, "y":2},
+                {"label":"K2B (B2,D0)", "x":11.75, "y":2},
+                {"label":"K2D (B2,D2)", "x":12.75, "y":2, "w":2.25},
+                {"label":"K2E (B2,D3)", "x":15, "y":2},
+                {"label":"K30 (B3,B5)", "x":0, "y":3, "w":1.25},
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/mechlovin9/info.json:55]: Fixing a couple errors with the key sizing/positioning.

image

```suggestion
                {"label":"K30 (B3,B5)", "x":0, "y":3, "w":2.25},
```

File: keyboards/mechlovin/hannah65/mechlovin9/info.json [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+                {"label":"K3A (B3,C7)", "x":10.25, "y":3},
+                {"label":"K3B (B3,D0)", "x":11.25, "y":3},
+                {"label":"K3C (B3,D1)", "x":12.25, "y":3, "w":1.75},
+                {"label":"K3D (B3,D2)", "x":14, "y":3},
+                {"label":"K3E (B3,D3)", "x":15, "y":3},
+                {"label":"K40 (B4,B5)", "x":0, "y":4, "w":1.25},
+                {"label":"K41 (B4,B6)", "x":1.25, "y":4, "w":1.25},
+                {"label":"K42 (B4,B7)", "x":2.5, "y":4, "w":1.25},
+                {"label":"K46 (B4,C3)", "x":3.75, "y":4, "w":6.25},
+                {"label":"K4B (B4,D0)", "x":11.25, "y":4, "w":1.25},
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah65/mechlovin9/info.json:73]:
```suggestion
                {"label":"K40 (B4,B5)", "x":0, "y":4, "w":1.5},
                {"label":"K41 (B4,B6)", "x":1.5, "y":4},
                {"label":"K42 (B4,B7)", "x":2.5, "y":4, "w":1.5},
                {"label":"K46 (B4,C3)", "x":4, "y":4, "w":7},
                {"label":"K4B (B4,D0)", "x":11, "y":4, "w":1.5},
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**